frontend/z3_axioms.py

```diff
@@ -16,12 +16,10 @@ def add(left, right, result, types):
     """
     return [
         Or(
-            And(types.subtype(left, types.num), types.subtype(right, types.num), types.subtype(result, types.num)),
-            And(types.subtype(left, types.seq), left == right, left == result)
+            And(types.subtype(left, types.seq), left == right, left == result),
+            And(types.subtype(left, types.num), types.subtype(right, left), result == left),
+            And(types.subtype(left, types.num), types.subtype(left, right), result == right),
```

Comment:
Should this maybe be And(types.subtype(right, types.num), types.subtype(left, right), result == right)?